server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -508,19 +511,15 @@ private ImmutableDruidDataSource getDataSource(final String dataSourceName)
       return null;
     }
 
-    Map<String, DataSegment> segmentMap = Maps.newHashMap();
+    SortedMap<String, DataSegment> segmentMap = new TreeMap<>();
     for (ImmutableDruidDataSource dataSource : dataSources) {
       Iterable<DataSegment> segments = dataSource.getSegments();
       for (DataSegment segment : segments) {
         segmentMap.put(segment.getIdentifier(), segment);
       }
     }
 
-    return new ImmutableDruidDataSource(
-        dataSourceName,
-        ImmutableMap.of(),
-        ImmutableMap.copyOf(segmentMap)
-    );
+    return new ImmutableDruidDataSource(dataSourceName, Collections.emptyMap(), segmentMap);
```

Comment:
Minor, I think: if we're building the `segmentMap` from scratch here anyway, might as well start it off as an ImmutableSortedMap. That way `ImmutableSortedMap.copyOfSorted` is a no-op and we avoid the extra copy.

Reply:
Fixed. Thanks.

Reply:
This causes the below error because the same segments can be added to the segmentMap multiple times. Reverted this change.

```
java.lang.IllegalArgumentException: Multiple entries with same key: datasource1_2010-01-22T00:00:00.000Z_2010-01-23T00:00:00.000Z_null=DataSegment{size=20, shardSpec=NoneShardSpec, metrics=[], dimensions=[], version='null', loadSpec=null, interval=2010-01-22T00:00:00.000Z/2010-01-23T00:00:00.000Z, dataSource='datasource1', binaryVersion='9'} and datasource1_2010-01-22T00:00:00.000Z_2010-01-23T00:00:00.000Z_null=DataSegment{size=20, shardSpec=NoneShardSpec, metrics=[], dimensions=[], version='null', loadSpec=null, interval=2010-01-22T00:00:00.000Z/2010-01-23T00:00:00.000Z, dataSource='datasource1', binaryVersion='9'}
	at io.druid.server.http.DatasourcesResourceTest.testSimpleGetTheDataSourceManyTiers(DatasourcesResourceTest.java:404)
```